### PR TITLE
chore: add unit tests

### PR DESCRIPTION
**Pull request by lgpage**
Closes: #143 
Fixes: #260

### CODE REVIEW

File: src/app/components/sandbox-controller/sandbox-controller.component.spec.ts [hunk 1]
```diff
@@ -1,29 +1,42 @@
...
   let component: SandboxControllerComponent;
   let fixture: ComponentFixture<SandboxControllerComponent>;
 
+  let executorServiceSpy: jasmine.SpyObj<ExecutorService>;
+  let streamBuilderServiceSpy: jasmine.SpyObj<StreamBuilderService>;
+  let inputStreamSpy: jasmine.SpyObj<InputStreamLike>;
+  let exampleSpy: jasmine.SpyObj<Example>;
+
   beforeEach(waitForAsync(() => {
+    inputStreamSpy = jasmine.createSpyObj<InputStreamLike>('InputStreamLike', ['updateNode']);
```
Comment (peterjokumsen) [inline src/app/components/sandbox-controller/sandbox-controller.component.spec.ts:26]: Name argument not really needed,

```suggestion
    inputStreamSpy = jasmine.createSpyObj<InputStreamLike>(['updateNode']);
```

File: src/app/components/side-nav/side-nav.component.spec.ts [hunk 2]
```diff
@@ -26,7 +34,14 @@ describe('SideNavComponent', () => {
       ],
       declarations: [SideNavComponent],
       providers: [
-        { provide: EXAMPLE, useClass: MockExample, multi: true },
+        ...exampleSpies.map((example) => ({ provide: EXAMPLE, useValue: example, multi: true })),
+        { provide: Router, useValue: routerSpy },
+        {
+          provide: RuntimeService, useValue: MockService(RuntimeService, {
```
Comment (peterjokumsen) [inline src/app/components/side-nav/side-nav.component.spec.ts:40]: to remove having to specify generic for `of` you could create a partial variable, i.e.

```typescript
const runtimeSvcOverriddes: Partial<RunTimeService> = {
  orientation$: of('landscape'),
  mediaSize$: of('large'),
}
```

And then:

```suggestion
          provide: RuntimeService,
          useValue: MockService(RuntimeService, runtimeSvOverrides),
```

File: src/app/components/sandbox-controller/sandbox-controller.component.spec.ts (hunk 1 above)
Comment (lgpage) [reply src/app/components/sandbox-controller/sandbox-controller.component.spec.ts:26]: Leaving as is. I have done this in too many places and changing it doesn't make any real difference.
Noted for future PRs :)

File: src/app/components/side-nav/side-nav.component.spec.ts (hunk 2 above)
Comment (lgpage) [reply src/app/components/side-nav/side-nav.component.spec.ts]: Updated

**Review by peterjokumsen — APPROVED** [review]

**Pull request merged**